### ISSUE TITLE
Updating a tag's colors does not reflect on the tag's page

### ISSUE THREAD

**Issue by vaidehijoshi** [forem/forem#12286]
**Describe the bug**

When I attempt to update the colors for a tag, the colors do not update on the tag's show page.

**To Reproduce**

1. Make sure that you are a Forem admin with _tag resource privileges_.
2. Create a new tag, and go to that tag's edit page (`/admin/tags/1/edit`, for example).
3. Change the colors of the tag at the bottom of the form (`Bg color hex` and `Text color hex`).
4. Save your changes!
5. Go to the tag's show page (`/t/my-new-tag`).
6. Observe that the colors you selected for your new tag are not showing up 😕 

**Note:** If you try _using_ that tag on a _post_, you will notice that the colors actually _are_ correct...so this seems specific to the tag's show page. I suspect that this has to do with how the bg color hex and text color hex are cached on the tag's show page. If I had to guess, I'd say that the cache is not being busted when it should be 😅 

**Expected behavior**

I would expect that when I update a tag, those updates are reflected whenever I look at the tag.

**Screenshots**

Here are some screenshots of me reproducing this on `forem.team` with the `#planning` tag.

1) Here are the colors I chose for my tag:
<img width="556" alt="Screen Shot 2021-01-14 at 12 09 01 PM" src="https://user-images.githubusercontent.com/6921610/104644834-25bfa600-5663-11eb-8759-53075f873628.png">

2) My chosen colors successfully seem to save and update the tag:
<img width="1327" alt="Screen Shot 2021-01-14 at 12 09 11 PM" src="https://user-images.githubusercontent.com/6921610/104644844-28ba9680-5663-11eb-8cfe-0666d24feda9.png">

3) When I go to the tag's show page, those colors are not rendering:
<img width="1394" alt="Screen Shot 2021-01-14 at 12 09 20 PM" src="https://user-images.githubusercontent.com/6921610/104644865-2e17e100-5663-11eb-9240-6e2101c71df7.png">

4) But when I look at _a post_ that uses my tag, the colors do render correctly and as expected:
<img width="919" alt="Screen Shot 2021-01-14 at 12 09 31 PM" src="https://user-images.githubusercontent.com/6921610/104644875-307a3b00-5663-11eb-91ea-3b1b836589c1.png">

**Comment by joaoGabriel55** [comment]
I got simulate this bug. I'm working on it

**Comment by vaidehijoshi** [comment]
> I got simulate this bug. I'm working on it

Sounds good, @joaoGabriel55! Thanks for picking this up. Let us know if you run into any issues or need any help :)

**Comment by joaoGabriel55** [comment]
I styled this way. This style/design is following your style guide?

![image](https://user-images.githubusercontent.com/23138717/104960295-b32c2e80-59b2-11eb-839b-8187dd62e98b.png)

Another color

![image](https://user-images.githubusercontent.com/23138717/104960581-48c7be00-59b3-11eb-939c-d64da9529d27.png)

**Comment by vaidehijoshi** [comment]
Hi, @joaoGabriel55! If you can please open a PR and link this issue, it will be much easier for our team to review the changes. Right now it's hard for me to leave any feedback without looking at the code itself :) We have more details on this in our [Contributing Guide](https://docs.forem.com/contributing/forem)

**Comment by joaoGabriel55** [comment]
Right, I will open a PR @vaidehijoshi.

**Comment by atsmith813** [comment]
I did some digging on this issue and discovered a few things.

**TLDR** - everything seems to be working as expected. However, we may want to consider changing some of our expectations. If we want to change any of these expectations, maybe we can close this issue and open up individual ones.

I'll address each spot individually.

----

<img width="1327" alt="Screen Shot 2021-01-14 at 12 09 11 PM" src="https://user-images.githubusercontent.com/6921610/104644844-28ba9680-5663-11eb-8cfe-0666d24feda9.png">
The tag up at the top of the page is updating as we expect 🎉 

----

<img width="1394" alt="tag_show_page" src="https://user-images.githubusercontent.com/15987080/105903924-813c4d00-5fee-11eb-88a0-ebb138ab6086.png">

1. The title is updating as we expect 🎉 . It's worth noting that this page is cached at the edge. When we update the tag itself, we [bust the cache for the tag asynchronously](https://github.com/forem/forem/blob/2fbc7c983e1d0a7bfb82051839fac60cf5ab02ae/app/models/tag.rb#L112). So while uncommon, it's _possible_ it takes a few moments to take effect depending on the Forem's background worker queue.
2. Currently, we don't really support [the custom colors in this location](https://github.com/forem/forem/blob/2fbc7c983e1d0a7bfb82051839fac60cf5ab02ae/app/views/articles/_single_story.html.erb#L58-L70). If the tag is a flare tag, then we show some more "style". If not, we just display the text by using the [`cached_tag_list_array`](https://github.com/forem/forem/blob/2fbc7c983e1d0a7bfb82051839fac60cf5ab02ae/app/decorators/article_decorator.rb#L20-L22). Since this is just an Array of Strings, we don't have the hex values available. We would need the values from the tag records themselves. There are definitely some performance implications we'll want to consider if/when we decide to change this since we'd be moving from an Array of cached Strings to (maybe) some queries.

----

<img width="919" alt="Screen Shot 2021-01-14 at 12 09 31 PM" src="https://user-images.githubusercontent.com/6921610/104644875-307a3b00-5663-11eb-91ea-3b1b836589c1.png">

This style of tag on the article itself does _not_ update. That's because when we update a tag, we only [bust the edge cache for tags](https://github.com/forem/forem/blob/master/app/services/edge_cache/bust_tag.rb) and not the articles tagged with that specific tag. I'm not sure if we would want to update this or not. I'm hesitant to update this because on [super popular tags on large Forems](https://dev.to/t/javascript), we would essentially be busting caches for tens of thousands of articles on _each_ update done to a tag. There are some performance implications we'll want to consider should we decide to support the behavior of having the tag style update right away.

**Comment by atsmith813** [comment]
Based on the feedback to [the findings above](https://github.com/forem/forem/issues/12286#issuecomment-767828348), I'm going to close this issue and re-open specific ones around the separate pieces of these quirks since there a few things at play that contribute to this UX 🚀
